libnetwork/ipam/allocator.go

```diff
@@ -41,18 +42,14 @@ func NewAllocator(lcAs, glAs []*net.IPNet) (*Allocator, error) {
 	return &a, nil
 }
 
-func newAddrSpace(predefined []*net.IPNet) (*addrSpace, error) {
-	pdf := make([]netip.Prefix, len(predefined))
-	for i, n := range predefined {
-		var ok bool
-		pdf[i], ok = toPrefix(n)
-		if !ok {
-			return nil, fmt.Errorf("network at index %d (%v) is not in canonical form", i, n)
-		}
-	}
+func newAddrSpace(predefined ipamutils.Subnetter) (*addrSpace, error) {
 	return &addrSpace{
-		subnets:    map[netip.Prefix]*PoolData{},
-		predefined: pdf,
+		subnets: map[netip.Prefix]*PoolData{},
+		predefined: map[ipamutils.IPVersion]*ipamutils.Subnetter{
+			ipamutils.IPv4: predefined.V4(),
```

Comment:
I'm not super happy with that: an address space with mixed address families feels weird. On the other hand `ReleasePool` currently relies on the pool's string identifier to determine what address space is involved.

@corhere @thaJeztah I can make a follow-up PR to make `addrSpace` handle a single address family if you think it'd be better.